resources/js/src/app/components/pageDesign/LanguageDetection.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+  <div v-if="languageText" class="row py-2">
+    <div class="col-md-8">{{ languageText }}</div>
+    <div class="col-md-4 text-right">
+      <a :href="languageRedirect" :class="'btn btn-sm btn-' + appearance">{{ buttonText }}</a>
+      <a href="#" @click="deactivateRedirection" class="m-sm-1"><i class="fa fa-close"></i></a>
+    </div>
+  </div>
+</template>
+
+<script>
+  export default {
+    data () {
+      return {
+        languageText: '',
+        buttonText: '',
+        languageRedirect: '',
+        textLanguages: ''
+      }
+    },
+    props: ["redirect", "appearance", "texttranslations", "buttontranslations"],
```

Comment:
Please use the object syntax to define the props of this component to have at least a type check on the prop values:

https://vuejs.org/v2/api/#props